src/abundances.jl

```diff
@@ -1,62 +1,49 @@
 # Methods for absolute and relative abundances
 
-struct AbundanceTable{T<:Real} <: AbstractArray{T,2}
-    table::Array{T,2}
-    samples::Vector{S} where S
-    features::Vector{R} where R
-end
-
-function AbundanceTable(df::DataFrame)
-    return AbundanceTable(
-            Matrix(df[:,2:end]),
-            names(df[2:end]),
-            Vector(df[1]))
-end
-function AbundanceTable(table::Array{T,2}) where T<:Real
-    return AbundanceTable(
-             table,
-             Vector{Int64}(1:size(table,2)),
-             Vector{Int64}(1:size(table,1)))
-end
-
-@forward_func AbundanceTable.table Base.getindex, Base.setindex, Base.length, Base.size
+abundancetable(df::DataFrame) = ComMatrix(df, sitecolumns = true)
+abundancetable(table::Array{T,2}, site = ["sample_$x" for x in indices(table, 2)],
+    species = ["feature_$x" for x in indices(table, 1)]) where T<:Real =
+    ComMatrix(table', species, site)
 
+@forward_func ComMatrix.occurrences Base.length
 
 """
 Filter an abundance table to the top `n` species accross all samples
 
-This function also adds a row for "other", which sums the
+This function also adds a row for "other", which sums the abundances of
+the remaining species
 """
-function filterabund(abun::AbundanceTable, n::Int=10)
+function filterabund(abun::AbstractComMatrix, n::Int=minimum(10, nspecies(abun)))
+    srt = sortperm(total_abundance_species(abun), rev=true)
+    keep, compact = srt[1:n], srt[n+1:end]
     # TODO: add prevalence filter
 
-    totals = [sum(abun[i,:]) for i in 1:size(abun, 1)]
-
-    srt = sortperm(totals, rev=true)
+    newabun = view(abun, species = keep)
+    remaining_abundances = total_abundance_site(view(abun, species = compact))
 
-    newabun = abun[srt[1:n], :]
-
-    remainder = [sum(abun[srt[n+1:end], i]) for i in 1:size(abun, 2)]'
-    newabun = vcat(newabun, remainder)
-    newrows = cat(1, abun.features[srt[1:n]], ["other"])
-
-    return AbundanceTable(newabun, abun.samples, newrows)
+    ComMatrix(hcat(newabun.occurrences, remaining_abundances), [specnames(newabun); "other"] , sitenames(abun))
 end
 
-filterabund(df::DataFrame, n::Int=10) = filterabund(AbundanceTable(df), n)
-
+filterabund(df::DataFrame, n::Int=10) = filterabund(abundancetable(df), n)
 
-function relativeabundance(a::AbundanceTable; kind::Symbol=:fraction)
+"""
+Express abundances as relative abundances, summing to 1 at each site (or to 100
+if `kind` is `:percent`)
+"""
+function relativeabundance!(abun::AbstractComMatrix{Float64}; kind::Symbol=:fraction)
     in(kind, [:percent, :fraction]) || error("Invalid kind: $kind")
 
-    relab = a.table
-    for j in 1:size(relab, 2)
-        s = sum(relab[:,j])
-        for i in 1:size(relab,1)
-            @inbounds relab[i,j] = a[i,j] / s
-        end
+    for j in 1:nspecies(abun)
```

Comment:
Just realized I screwed up with the transposed matrix here - it has to be the other way round. I'll fix this but wait with pushing the fix so as not to make all my comments as "outdated"